src/lib.rs

```diff
@@ -112,3 +115,60 @@ impl PrivilegeLevel {
         }
     }
 }
+
+/// A wrapper that can be used to safely create one mutable reference `&'static mut T` from a static variable.
+///
+/// `Singleton` is safe because it ensures that it only ever gives out one reference.
+///
+/// ``Singleton<T>` is a safe alternative to `static mut` or a static `UnsafeCell<T>`.
+#[derive(Debug)]
+pub struct Singleton<T> {
```

Comment:
I think we should choose a different name for this because "singleton" is commonly used to describe classes/types that only have a single instance. With this wrapper, it is fine to create multiple `&'static mut T` by using multiple statics. So maybe something like `SingleUseCell`?

Reply:
Agreed, I like the name `SingleUseCell`.